### DISCUSSION TITLE
Translations - Announcements, feedback, etc.

### DISCUSSION THREAD

**Discussion by deckerst** [deckerst/aves#152] [General]
> In general, translating everything all the time might be counterproductive, because I often add, modify and remove translation keys during active development.

> For people who help with translations but do not open PRs, I simply email them a few days before releasing a new version, keeping the noise to a minimum.

> Not sure what would work best for translations via PR. Maybe I could open a permanent issue/discussion for announcements. You could subscribe to that issue/discussion, and be informed when I'm about to release a new version. So you would not need to manually check the "untranslated.json" file by yourself.

Here it is.

Edit: [guide to contribute to translations](https://github.com/deckerst/aves/wiki/Contributing-to-Translations)

**Comment by n-berenice** [comment]
Ok I've set it to receive all notifications, they ought to work now.

**Comment by n-berenice** [comment]
Ok so I was doing some actual testing with the latest apk on an actual device and I've found some typos in app_es.arb in the following lines:

```
466 "settingsRemoveAnimationsTitle": "Remover animaciones",
467 "settingsTimeToTakeActionTile": "Hora de entrar en acción", 
468 "settingsTimeToTakeActionTitle": "Hora de entrar en acción",
```

it should be 
```
466 "settingsRemoveAnimationsTitle": "Remover animaciones",
467 "settingsTimeToTakeActionTile": "Retraso para ejecutar una acción",
468 "settingsTimeToTakeActionTitle": "Retraso para ejecutar una acción",
```

That's all for now.

**Reply by deckerst** [reply]
Thanks, I committed the fix. In the future, I might not reply and instead just add a thumb up to that kind of correction message, to limit spam notifications.

Also, I just put on the wiki a [guide to contribute to translations](https://github.com/deckerst/aves/wiki/Contributing-to-Translations). Let me know if during the translation you found some pain points that should be addressed by the guide.

**Reply by n-berenice** [reply]
timetotakeaction was painful enough because I'm running an older version of android and I had no idea that was a setting, tbh I thought it was related to video edition

**Comment by deckerst** [comment]
@n-berenice I automated screenshot generation and localization so they're all consistent, based on the same dataset. You can check the [Spanish ones for IzzyOnDroid](https://github.com/deckerst/aves/tree/develop/fastlane/metadata/android/es-MX/images/phoneScreenshots). I guess they will be picked up next time there's a release.

**Comment by deckerst** [comment]
@n-berenice I will release a new version of Aves in the coming days. Now is a good time for translating [the missing keys](https://github.com/deckerst/aves/blob/develop/untranslated.json#L13). By the way, I already contacted the other translators who prefer email.

**Reply by n-berenice** [reply]
Hey I was away for a few weeks. I've edited the file already and pushed it to my fork. Should I open a PR? B.

**Reply by deckerst** [reply]
Sure, that would be nice, thanks!

**Comment by n-berenice** [comment]
Note: pushed changes related to video controls to local fork, _rocket emoji_

**Reply by deckerst** [reply]
That's cool, but be aware that all related keys are subject to change during active development. Especially this, I'm experimenting a lot.

**Reply by deckerst** [reply]
I didn't want to wait too much for the release of v1.6.2, so I took the liberty of including the missing translations from your local branch.

**Reply by n-berenice** [reply]
> I didn't want to wait too much for the release of v1.6.2, so I took the liberty of including the missing translations from your local branch.

I was wondering why they weren't there anymore. PR open.

**Comment by deckerst** [comment]
@n-berenice I will release a new version of Aves in the coming days. Now is a good time for translating the missing keys in Spanish.

**Reply by n-berenice** [reply]
I'm having technical issues again. 

I can't _fetch upstream_ to update my fork and start editing the files because it's telling me there are unresolved conflicts and asks me to open a PR, I do that and assign myself believing it'll allow me to solve said conflicts but instead asks me to start using terminal commands or open it using GitHub desktop, I choose the latter and it tells me "a branch with that name already exists", like... what branch? there's literally just one branch, just wonderful, exactly what I needed on a weekend.
I think I'll have to delete everything and fork the entire project again, if that's even possible.

**Reply by n-berenice** [reply]
Update: I did a thing called rebase that supposedly overwrites every conflicting file on my fork and I believe it worked as the `untranslated.json` has changed.

**Reply by deckerst** [reply]
Glad to hear you worked it out. Conflicts can be intimidating if you're not familiar with the tools.

I sometimes edit the localization files (to remove obsolete keys, fix formatting, etc.) so always pull from upstream before you edit something. This will reduce the chance of conflicts.

I just merged your PR (thanks!) and re-committed my formatting changes. You could pull from upstream now.

**Comment by deckerst** [comment]
@n-berenice @glemco I will release a new version of Aves in the coming days. Now is a good time for translating the missing keys in Spanish and Italian. You can see which ones are missing [here](https://github.com/deckerst/aves/blob/develop/untranslated.json) and the reference English is [there](https://github.com/deckerst/aves/blob/develop/lib/l10n/app_en.arb).

And don't forget to pull from upstream *before* editing the translation!

**Comment by deckerst** [comment]
@n-berenice @glemco I will release a new version of Aves in the coming days. Now is a good time for translating the missing keys in Spanish and Italian. There are just 2 this time:
```
"settingsShowBottomNavigationBar": "Show bottom navigation bar",
"settingsThumbnailShowTagIcon": "Show tag icon",
```

**Comment by deckerst** [comment]
@n-berenice @glemco @metezd I will release a new version of Aves in the coming days. Now is a good time for translating the missing keys in Spanish, Italian and Turkish. You can see which ones are missing [here](https://github.com/deckerst/aves/blob/develop/untranslated.json) and the reference English is [there](https://github.com/deckerst/aves/blob/develop/lib/l10n/app_en.arb).

And don't forget to pull from upstream *before* editing the translation!

**Comment by deckerst** [comment]
@n-berenice @glemco @metezd I will release a new version of Aves in the coming days. Now is a good time for translating the missing keys in Spanish, Italian and Turkish. There are just 5 this time:
```json
"filterOnThisDayLabel": "On this day",
"settingsSlideshowFillScreen": "Fill screen",
"settingsScreenSaverPageTitle": "Screen Saver",
"settingsWidgetPageTitle": "Photo Frame",
"settingsWidgetShowOutline": "Outline",
```

And don't forget to pull from upstream *before* editing the translation!
@n-berenice if you keep running into problems with git, don't hesitate to just [mail the translation to me](mailto:gallery.aves@gmail.com).

**Reply by n-berenice** [reply]
I've managed to sync everything, solve all conflicts and now after I've spent an hour editing the files it says "Authentication failed." when I try to push my changes, lovely, just lovely.

I'll leave the modified app_es.arb strings here for now.
```
"filterOnThisDayLabel": "De este día",
"settingsSlideshowFillScreen": "Llenar pantalla",
"settingsScreenSaverPageTitle": "Protector de pantalla",
"settingsWidgetPageTitle": "Marco de foto",
"settingsWidgetShowOutline": "Borde",
"settingsViewerGestureSideTapNext": "Toque en los bordes de la pantalla para mostrar el elemento anterior/siguiente",
```

**Comment by deckerst** [comment]
@n-berenice @glemco @metezd I will release a new version of Aves in the coming days. Now is a good time for translating the missing keys in Spanish, Italian and Turkish. You can see which ones are missing [here](https://github.com/deckerst/aves/blob/develop/untranslated.json) and the reference English is [there](https://github.com/deckerst/aves/blob/develop/lib/l10n/app_en.arb).

And don't forget to pull from upstream *before* editing the translation!

**Comment by deckerst** [comment]
@n-berenice @glemco @metezd I will release a new version of Aves in the coming days. Now is a good time for translating the missing keys in Spanish, Italian and Turkish. You can see which ones are missing [here](https://github.com/deckerst/aves/blob/develop/untranslated.json) and the reference English is [there](https://github.com/deckerst/aves/blob/develop/lib/l10n/app_en.arb).

And don't forget to pull from upstream *before* editing the translation!

**Comment by deckerst** [comment]
@n-berenice @glemco I will release a new version of Aves in the coming days. Now is a good time for translating the missing keys in Spanish and Italian. You can see which ones are missing [here](https://github.com/deckerst/aves/blob/develop/untranslated.json) and the reference English is [there](https://github.com/deckerst/aves/blob/develop/lib/l10n/app_en.arb).

And don't forget to pull from upstream *before* editing the translation!

**Comment by deckerst** [comment]
@n-berenice @glemco I will release a new version of Aves in the coming days. But from now on, translations are powered by [Weblate](https://hosted.weblate.org/projects/aves/).

Here's the [main component for Spanish](https://hosted.weblate.org/projects/aves/app-main/es/), and [same for Italian](https://hosted.weblate.org/projects/aves/app-main/it/). Other contributors already filled the gaps in these languages, but you may be interested in reviewing/suggesting translations, or monitoring that space for the future.

I won't notice upcoming releases here anymore, but anybody interested in translating and having questions on the topic are welcome to ask them in this thread. Don't hesitate to let me know ways to make things more convenient for translators on Weblate.

**Reply by glemco** [reply]
I'm always struggling finding things on weblate, how can I accept a suggestion (If I can at all)?

**Reply by deckerst** [reply]
I don't think you can, as I'm the only admin and there's no other user with elevated privileges. If you'd like to be able to manage Italian translations, i could try and add you to the project, with some extra rights. I don't know well Weblate, I'm discovering it as needed. Please let me know your weblate user handle if you're interested.

**Reply by glemco** [reply]
Well, I think it would be easier for you if you put us as contributors also there, you may manage with Italian but Turkish, Korean and so on are another story 😅
It should be glemco also there, let me know if you can't find me.

**Reply by deckerst** [reply]
I will. But i want to clarify something. People, without extra right, can already provide a translation for an untranslated key. And they can also provide suggestions to approved translations. What they cannot do is overwrite an approved translation. Many people already contributed to translations on weblate without being added as project contributors with special rights.

By the way, i don't know Turkish but i do know Korean. I spent more than a decade in Korea ^^

**Reply by glemco** [reply]
Well, in the end it's up to you, it's your project, I'd just be happy to help ;)
Uh, chapeau!

**Reply by deckerst** [reply]
@glemco I gave you power user rights for Italian, so you can manage suggestions, modify approved translations, etc.

**Comment by rehork** [comment]
@deckerst Thank you for developing this application.
 I translated quite some strings. However, there are strings which aren't fully visible on some phones like 5'', etc. For instance in section Settings -> Video -> Controls -> Buttons -> Play & seek backward/forward
is not even fully visible in English. 
I'm not sure if it's a challenging task for you, but I would suggest two line text in such menus or automatically scrolling text when not fully visible in a window.

**Reply by deckerst** [reply]
Right, it's something I wanted to fix eventually. Thanks for reminding me.

And thanks for your translations! You've probably seen it but Polish made it to the recently released Aves v1.7.9!

**Reply by rehork** [reply]
Yes, I've seen the translation in the latest release. That's why I've noticed the issue with not fully visible texts. Thanks anyway!
